Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -296,14 +291,37 @@ bool PPCSymbolDB::LoadMap(const std::string& filename, bool bad)
     if (section_name.empty())
       continue;
 
-    // Detect two columns with three columns fallback
+    // Column detection heuristic
     if (column_count == 0)
     {
-      const std::string_view stripped_line = StripSpaces(line);
-      if (std::count(stripped_line.begin(), stripped_line.end(), ' ') == 1)
-        column_count = 2;
-      else
+      constexpr auto is_hex_str = [](const std::string& s) {
+        return !s.empty() && s.find_first_not_of("0123456789abcdefABCDEF") == std::string::npos;
+      };
+      const std::string stripped_line(StripSpaces(line));
+      std::istringstream oss(stripped_line);
+      std::string word;
+
+      // Two columns format:
+      // 80004000 zz_80004000_
+      if (!(oss >> word) || word.length() != 8 || !is_hex_str(word))
+        continue;
+      column_count = 2;
```

Comment:
Would it be faster to tokenize the string by spaces, then later validate whether they are hex or not?

Reply:
I don't think it will be faster since we don't need to tokenize the whole line if the first word isn't a hexstring. If I understand correctly it will also require a container like a vector to store the string tokens and then process them which might be slower.